.github/workflows/main.yml

```diff
@@ -68,7 +73,22 @@ jobs:
     - name: Install testing dependencies
       run: python -m pip install tox codecov
     - name: Run tests with ${{ matrix.name }}
+      if: ${{ !contains(matrix.toxenv,'-casa') }}
       run: tox -v -e ${{ matrix.toxenv }}
+    - name: Run tests with ${{ matrix.name }} (incl. CASA)
+      if: ${{ contains(matrix.toxenv,'-casa') }}
+      run: |
+          pip install casaconfig setuptools
+          mkdir ${GITHUB_WORKSPACE}/.casa
+          mkdir ${GITHUB_WORKSPACE}/.casa/data
+          chmod 777 -R ${GITHUB_WORKSPACE}/.casa
+          echo "datapath=['${GITHUB_WORKSPACE}/.casa/data']; measurespath=datapath[0]; measures_auto_update=True; data_auto_update=True" > $GITHUB_WORKSPACE/config.py
+          pip install casadata
+          # https://casadocs.readthedocs.io/en/stable/notebooks/external-data.html#Populating-the-Data-Directory-Manually
+          python -m casaconfig --update-all
+          python -m casaconfig --current-data
+          CASASITECONFIG=$GITHUB_WORKSPACE/config.py tox -v -e ${{ matrix.toxenv }}
```

Comment:
@e-koch - why is this whole block needed as opposed to dealing with it inside the tox config? Does it not work that way?

Reply:
I mostly copied this over from spectral-cube after @keflavich 's several tests to get a working version with casa's need to grab external data on startup.

@keflavich do you recall testing whether we could put this in tox?

Reply:
I'll take a quick look to see if it can be done in tox here